src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -459,7 +417,13 @@ && shouldStopProcessing(
                 CompletableFuture.supplyAsync(
                     () -> {
                       try {
-                        BlobMetadata blobMetadata = buildAndUpload(blobPath, blobData);
+                        // Get the channel flush context from the first channel in the blob. This
+                        // only matters when the client is in Iceberg mode. In Iceberg mode, all
+                        // channels in the blob belong to the same table.
+                        ChannelFlushContext channelFlushContext =
+                            blobData.get(0).get(0).getChannelContext();
+                        BlobMetadata blobMetadata =
+                            buildAndUpload(blobPath, blobData, channelFlushContext);
```

Comment:
again, looks like all we need is the table name?

Reply:
Changed signature.